### DISCUSSION TITLE
on demand ISR does not support custom server with https

### DISCUSSION THREAD

**Discussion by aNyMoRe0505** [vercel/next.js#34664] [Help] [answered]
Hi, next.js team
Thanks for on-demand (manually) revalidation feature, it is amazing!

I tried it immediately with res.unstable_revalidate api. It works in smooth way. But I found that it not works for custom server with https
you can check this [repo](https://github.com/aNyMoRe0505/on-demand-isr-with-custom-server), before you start the server, you need to use `mkcert` to generate credential files, then execute `npm run dev`
then you can call revalidate api (https://localhost:3000/api/revalidate), you will get the error message in terminal `Error: Failed to revalidate /`

Once you replace `https` with `http` (you can check `server.js`), everything work fine including revalidate page manually

Is it intentional or I just miss something ? Thanks!

**Comment by TimoWilhelm** [comment, marked as answer]
This actually started working for me with the latest next canary release (`^12.1.1-canary.15` in my case). Seems to be related to PR https://github.com/vercel/next.js/pull/35386.

**Reply by aNyMoRe0505** [reply]
thanks
